### DISCUSSION TITLE
Why the usual import "import aiohttp" causes a 10-15 MB increase in RAM

### DISCUSSION THREAD

**Discussion by masksshow** [aio-libs/aiohttp#5889] [Q&A] [answered]
Hello.

Why the usual import "import aiohttp" causes a 10-15 MB increase in RAM. This is critical for me.
I have implemented several projects using the aiohttp client. I've always used docker.
1.What is it connected with?
2. Is it possible to reduce the consumption of RAM.

**Comment by Dreamsorcerer** [comment, marked as answer]
I've no idea, but you could try using tracemalloc to investigate: https://docs.python.org/3/library/tracemalloc.html
If you find any obvious issues/improvements, then we can look at fixing them (I doubt any maintainer currently has time to do this investigation).

Also, have you tried only importing the client side? e.g. with `from aiohttp import ClientSession`, not sure if that might give an improvement or not...

**Reply by webknjaz** [reply]
This won't change a thing since Python's import machinery will first import the top-level `__init__.py` before going deeper.

**Reply by Dreamsorcerer** [reply]
I thought it was a bit a stretch, but wondered if maybe unused items would then get discarded and garbage collected or something, as they have no references outside the init file.
